### PR TITLE
Updated configmap mountpoints based on ETCD replicas.

### PR DESCRIPTION
**Pull request by abdasgupta**
**How to categorize this PR?**

/area TODO
/kind TODO

**What this PR does / why we need it**:
This PR is rework of #193. 
**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Release note**:

**breaking**
```other operator
- If Spec.Replicas in ETCD CR is greater than 0 and a even number, then no statefulset for ETCD nodes will be created and so the ETCD cluster won't be setup by Druid
```

```other operator
- Deployed configmap programmatically as component instead of chart
- configmap configures ETCD config based on the number of nodes in cluster. Number of nodes in cluster is derived from spec.Replicas of ETCD CR
- Introduced separate TLS config for client and peer communication with ETCD cluster
```

### CODE REVIEW

File: config/rbac/leader_election_role.yaml [hunk 1]
```diff
@@ -5,7 +5,7 @@ metadata:
   name: leader-election-role
 rules:
 - apiGroups:
-  - ""
+  - "coordination.k8s.io
```
Comment (aaronfern) [inline config/rbac/leader_election_role.yaml:8]:
```suggestion
  - "coordination.k8s.io"
```

**Review by aaronfern — CHANGES_REQUESTED** [review]
Can you please remove `charts/etcd/templates/etcd-configmap.yaml` as it's not required anymore

File: config/rbac/leader_election_role.yaml (hunk 1 above)
Comment (ishan16696) [reply config/rbac/leader_election_role.yaml:8]: are you sure `coordination.k8s.io` about this as when I tested druid before release I didn’t face any error.

File: controllers/etcd_controller.go [hunk 2]
```diff
@@ -940,21 +823,15 @@ func (r *EtcdReconciler) reconcileRoleBinding(ctx context.Context, logger logr.L
...
-		Lease:   componentlease.GenerateValues(etcd),
-		Service: componentservice.GenerateValues(etcd),
-	}
-
-	values, err := getMapFromEtcd(ctx, r.Client, r.ImageVector, etcd, val, r.disableEtcdServiceAccountAutomount)
-
-	if err != nil {
-		return nil, nil, err
+	// Check if Spec.Replicas is odd or even.
+	if etcd.Status.Replicas > 1 && etcd.Spec.Replicas&1 == 0 {
```
[the remark this answers is not in the document]
Comment (abdasgupta) [reply controllers/etcd_controller.go]: Yes. I already realised it. Fixed it.

File: config/rbac/leader_election_role.yaml (hunk 1 above)
Comment (abdasgupta) [reply config/rbac/leader_election_role.yaml]: Fixed

**Review by aaronfern — APPROVED** [review]
Thanks for the PR!
/lgtm

**Pull request merged**